scipy/signal/fir_filter_design.py

```diff
@@ -632,13 +637,24 @@ def firwin2(numtaps, freq, gain, nfreqs=None, window='hamming', nyq=None,
     if nfreqs is None:
         nfreqs = 1 + 2 ** int(ceil(log(numtaps, 2)))
 
+    d = np.diff(freq)
+    if (d == 0).any():
+        freq = np.array(freq, copy=True)
+
     # Tweak any repeated values in freq so that interp works.
-    eps = np.finfo(float).eps
+    eps = np.finfo(float).eps * nyq
     for k in range(len(freq)):
         if k < len(freq) - 1 and freq[k] == freq[k + 1]:
             freq[k] = freq[k] - eps
             freq[k + 1] = freq[k + 1] + eps
 
+    # Check if freq is strictly increasing after tweak
+    d = np.diff(freq)
+    if (d <= 0).any():
+        raise ValueError("freq cannot contain numbers that are too close "
+                         "(within eps * nyquist: "
```

Comment:
Since the docs deprecate the specification of `nyq`, preferring `fs`, the error messages should communicate in `fs` as well. So, something like `eps * (fs/2)` here.